eZ/Publish/Core/REST/Server/Input/Parser/ViewInput.php

```diff
@@ -36,19 +37,28 @@ public function parse( array $data, ParsingDispatcher $parsingDispatcher )
         $restViewInput = new RestViewInput();
 
         // identifier
-        if ( !array_key_exists( 'identifier', $data ) )
-        {
+        if (!array_key_exists( 'identifier', $data )) {
             throw new Exceptions\Parser( "Missing <identifier> attribute for <ViewInput>." );
         }
-        $restViewInput->identifier  = $data['identifier'];
+        $restViewInput->identifier = $data['identifier'];
 
         // query
-        if ( !array_key_exists( 'Query', $data ) || !is_array( $data['Query'] ) )
-        {
+        if (!array_key_exists( 'Query', $data ) || !is_array( $data['Query'] )) {
             throw new Exceptions\Parser( "Missing <Query> attribute for <ViewInput>." );
         }
 
-        $query = new Query();
+        if ( !isset( $data['Query']['_type'] ) || $data['Query']['_type'] == 'content' )
+        {
+            $query = new Query();
+        }
+        else if ( isset( $data['Query']['_type'] ) || $data['Query']['_type'] == 'location' )
```

Comment:
I wasn't suggesting this because of micro optimization, I was optimizing for reading as I personally read if sentences top to down.